airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt

```diff
@@ -295,19 +297,32 @@ open class S3StorageOperations(
         cleanUpBucketObject(objectPath, listOf())
     }
 
-    private fun listObjects(objectPath: String): ObjectListing {
+    private fun listObjectsByPage(
```

Comment:
Should this be `forObjectByPage` now?

Reply:
👍  sounds more right since it has consumer semantics rather than return value.